crates/notion-core/src/inventory/mod.rs

```diff
@@ -604,45 +619,81 @@ fn resolve_node_versions(url: &str) -> Fallible<serial::NodeIndex> {
         Some(serial) => Ok(serial),
         None => {
             let spinner = progress_spinner(&format!("Fetching public registry: {}", url));
+
             let mut response: reqwest::Response =
                 reqwest::get(url).with_context(registry_fetch_error("Node", url))?;
-            let response_text: String = response.text().unknown()?;
-            let cached: NamedTempFile = NamedTempFile::new_in(path::tmp_dir()?).unknown()?;
+            let response_text = response
+                .text()
+                .with_context(registry_fetch_error("Node", url))?;
+            let index: serial::NodeIndex =
+                serde_json::de::from_str(&response_text).with_context(|_| {
+                    ErrorDetails::ParseNodeIndexError {
+                        from_url: url.to_string(),
+                    }
+                })?;
+
+            let tmp_root = path::tmp_dir()?;
+            // Helper to lazily determine temp dir string, without moving the file into the closures below
+            let get_tmp_root = || tmp_root.to_string_lossy().to_string();
+
+            let cached: NamedTempFile = NamedTempFile::new_in(&tmp_root).with_context(|_| {
+                ErrorDetails::CreateTempFileError {
+                    in_dir: get_tmp_root(),
+                }
+            })?;
 
             // Block to borrow cached for cached_file.
             {
                 let mut cached_file: &File = cached.as_file();
-                cached_file.write(response_text.as_bytes()).unknown()?;
+                cached_file
+                    .write(response_text.as_bytes())
+                    .with_context(|_| ErrorDetails::WriteNodeIndexCacheError {
+                        file: cached.path().to_string_lossy().to_string(),
+                    })?;
             }
 
             let index_cache_file = path::node_index_file()?;
             ensure_containing_dir_exists(&index_cache_file)?;
-            cached.persist(index_cache_file).unknown()?;
+            cached.persist(&index_cache_file).with_context(|_| {
+                ErrorDetails::WriteNodeIndexCacheError {
+                    file: index_cache_file.to_string_lossy().to_string(),
+                }
+            })?;
 
-            let expiry: NamedTempFile = NamedTempFile::new_in(path::tmp_dir()?).unknown()?;
+            let expiry: NamedTempFile = NamedTempFile::new_in(&tmp_root).with_context(|_| {
```

Comment:
Looks like the whole module has a lot of this; let's tackle that as a small follow-on PR.

Reply:
Good call, filed an issue #375